exploration/demo/evaluate.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+
+import polars as pl
+
+from linmod.eval.proportions import mae
+
+data = (
+    pl.scan_csv(Path("../../data/metadata.csv"))
+    .cast({"date": pl.Date}, strict=False)
```

Comment:
I think the better think here is `try_parse_dates=True` in the `scan_csv()`

Reply:
It would be cleaner, but there is actually a functionality difference that makes me still consider the current method...

We want to ensure the date column sets non-compliant entries (for example, dates that are only month+year) to `null`. This is important, because I use the null filter to remove rows that don't have dates resolved to the day. `cast(..., strict=False)` does this.

`scan_csv(..., try_parse_dates=True)` fails in this scenario without also setting `ignore_errors=True`, which scares me

Reply:
I think I should document this somewhere, even as a comment. But I will probably do that when I move this date filtering into the global preprocessing script. At that point, a user loading the output of that script could totally use `try_parse_dates=True` without fail, I think

Reply:
interesting. agree, this could be documented (or even written as a separate function)

Reply:
https://github.com/CDCgov/cfa-viral-lineage-model/blob/8dd8b0ccbcec850e768a63b724fcb30d0263e865/linmod/data.py#L160-L163